src/main/kotlin/no/nav/familie/ks/sak/kjerne/tilbakekreving/FagsystemsbehandlingService.kt

```diff
@@ -0,0 +1,77 @@
+import no.nav.familie.ks.sak.kjerne.behandling.BehandlingService
+import no.nav.familie.ks.sak.kjerne.behandling.domene.Behandling
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vedtak.VedtakService
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.PersonopplysningGrunnlagService
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+
+@Service
+class FagsystemsbehandlingService(
+    private val behandlingService: BehandlingService,
+    private val personopplysningGrunnlagService: PersonopplysningGrunnlagService,
+    private val arbeidsfordelingService: ArbeidsfordelingService,
+    private val vedtakService: VedtakService,
+    private val tilbakekrevingsbehandlingHentService: TilbakekrevingsbehandlingHentService,
+    private val kafkaProducer: KafkaProducer
+) {
+
+    fun hentFagsystemsbehandling(request: HentFagsystemsbehandlingRequest): HentFagsystemsbehandlingRespons {
+        logger.info("Henter behandling for behandlingId=${request.eksternId}")
+        val behandling = behandlingService.hentBehandling(request.eksternId.toLong())
+
+        return lagRespons(request, behandling)
+    }
+
+    fun sendFagsystemsbehandling(
+        respons: HentFagsystemsbehandlingRespons,
+        key: String,
+        behandlingId: String
+    ) {
+        kafkaProducer.sendFagsystemsbehandlingRespons(respons, key, behandlingId)
+    }
+
+    private fun lagRespons(
+        request: HentFagsystemsbehandlingRequest,
+        behandling: Behandling
+    ): HentFagsystemsbehandlingRespons {
+        val behandlingId = behandling.id
+        val persongrunnlag = personopplysningGrunnlagService.hentAktivPersonopplysningGrunnlagThrows(behandlingId = behandlingId)
+        val arbeidsfordeling = arbeidsfordelingService.hentArbeidsfordelingPåBehandling(behandlingId)
+        val aktivVedtak = vedtakService.hentAktivVedtakForBehandling(behandlingId)
+
+        val faktainfo = Faktainfo(
+            revurderingsårsak = behandling.opprettetÅrsak.visningsnavn,
+            revurderingsresultat = behandling.resultat.displayName,
+            tilbakekrevingsvalg = tilbakekrevingsbehandlingHentService.hentTilbakekreving(behandlingId).valg
+        )
+
+        val hentFagsystemsbehandling = HentFagsystemsbehandling(
+            eksternFagsakId = request.eksternFagsakId,
+            eksternId = request.eksternId,
+            ytelsestype = request.ytelsestype,
+            regelverk = behandling.kategori.tilRegelverk(),
+            personIdent = behandling.fagsak.aktør.aktivFødselsnummer(),
+            språkkode = persongrunnlag.søker.målform.tilSpråkkode(),
+            enhetId = arbeidsfordeling.behandlendeEnhetId,
+            enhetsnavn = arbeidsfordeling.behandlendeEnhetNavn,
+            revurderingsvedtaksdato = aktivVedtak.vedtaksdato?.toLocalDate()
+                ?: throw Feil("Finnes ikke vedtaksdato for behandling $behandlingId"),
```

Comment:
Kan nullsjekken utføres rett etter vi henter ut vedtak for å unngå unødig kall til tilbakekreving (linje 53) hvis den mangler vedtaksdato?

Reply:
Siden vi bare er interessert i vedtaksdatoen så kan man kanskje bare trekke dette ut med engang slik at

val aktivVedtak = vedtakService.hentAktivVedtakForBehandling(behandlingId)

gjøres om til

val aktivVedtaksdato = aktivVedtak.vedtaksdato?.toLocalDate()
                 ?: throw Feil("Finnes ikke vedtaksdato for behandling $behandlingId")